### PR TITLE
ISSD-3244 [PRM] Add partner roles to Partner Portal's auto-provisioning

### PR DESCRIPTION
**Pull request by thaynam**

### CODE REVIEW

**Review by jairmedeiros — APPROVED** [review]

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 1]
```diff
@@ -787,12 +845,24 @@ private void _updateAccountRegion(
...
 	private static final String _KORONEIKI_ROLE_PARTNER_MANAGER_NAME =
 		"Partner Manager";
 
 	private static final String _KORONEIKI_ROLE_PARTNER_MEMBER_NAME =
 		"Partner Member";
 
+	private static final String _KORONEIKI_ROLE_PARTNER_SALES_USER_NAME =
+		"Partner Sales User";
+
+	private static final String _KORONEIKI_ROLE_TECHNICAL_USER_NAME =
```
Comment (amosfong) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:863]: Can you make this var name PARTNER_TECHNICAL_USER to match the actual name better

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 2]
```diff
@@ -787,12 +845,24 @@ private void _updateAccountRegion(
 	private static final String _ACCOUNT_ROLE_PARTNER_SALES_USER_NAME =
 		"[Account] Partner Sales User (PSU)";
 
+	private static final String _ACCOUNT_ROLE_TECHNICAL_SALES_USER_NAME =
+		"[Account] Partner Technical User (PTU)";
+
+	private static final String _KORONEIKI_ROLE_MARKETING_USER_NAME =
```
Comment (amosfong) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:851]: can you name this PARTNER_MARKETING_USER_NAME

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 3]
```diff
@@ -186,6 +186,35 @@ else if (contactRoleName.equals(_KORONEIKI_ROLE_PARTNER_MEMBER_NAME)) {
 			_assignUserToRegularRole(
 				contactEmailAddress, _ROLE_PARTNER_SALES_USER_NAME);
 		}
+		else if (contactRoleName.equals(_KORONEIKI_ROLE_MARKETING_USER_NAME)) {
```
Comment (amosfong) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:189]: Can you sort these with the other if statements?

**Review by amosfong — APPROVED** [review]

**Comment by amosfong** [comment]
ci:forward:force

**Comment by amosfong** [comment]
ci:forward:force

**Comment by jpince** [comment]
Errors are unrelated, Brian fixed this in upstream https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/upgrade/packageinfo. Thaynam will manually forward to Brian.

**Comment by thaynam** [comment]
[manually forwarded to Brian](https://github.com/brianchandotcom/liferay-portal/pull/148625)

**Pull request closed without merging**